entproto/cmd/protoc-gen-entgrpc/converter.go

```diff
@@ -65,7 +66,21 @@ func (g *serviceGenerator) newConverter(fld *entproto.FieldMappingDescriptor) (*
 	if fld.IsEdgeField {
 		efld = fld.EntEdge.Type.ID
 	}
+
 	switch {
+	case efld.IsUUID() && efld.Type.ValueScanner():
+		out.toEntScannerConstructor = protogen.GoImportPath("github.com/google/uuid").Ident("UUID")
+	case efld.HasGoType() && efld.Type.ValueScanner():
+		// Ident returned from ent already has the packagename prefixed. Strip it since `g.QualifiedGoIdent`
+		// adds it back.
+		split := strings.Split(efld.Type.Ident, ".")
+		out.toEntScannerConstructor = protogen.GoIdent{GoName: split[1], GoImportPath: protogen.GoImportPath(efld.Type.PkgPath)}
+	case efld.IsBool() && efld.Type.ValueScanner():
+		out.toEntScannerConversion = "bool"
+	case efld.IsBytes() && efld.Type.ValueScanner():
+		out.toEntScannerConversion = "[]byte"
+	case efld.IsString() && efld.Type.ValueScanner():
+		out.toEntScannerConversion = "string"
```

Comment:
Please replace all the `efld.Type.ValueScanner()` checks with one check. 
```suggestion
	switch {
	case !efld.Type.ValueScanner():
		switch {
		case efld.IsBool(), efld.IsBytes(), efld.IsString():
		case efld.Type.Numeric():
			// ..
		default:
			// ..
		}
	case efld.IsUUID():
		// ...
	case efld.HasGoType():
		// ...
	case efld.IsBool():
		out.toEntScannerConversion = "bool"
	case efld.IsBytes():
		out.toEntScannerConversion = "[]byte"
	case efld.IsString():
		out.toEntScannerConversion = "string"
	}
```